src/interdiff.c

```diff
@@ -1070,7 +1157,23 @@ trim_context (FILE *f /* positioned at start of @@ line */,
 			if (total_count < strip_post)
 				continue;
 
-			fwrite (line, (size_t) got, 1, out);
+			{
+				enum line_type type;
+				switch (line[0]) {
+				case '+':
+					type = LINE_ADDED;
+					break;
+				case '-':
+					type = LINE_REMOVED;
+					break;
+				case ' ':
+				case '\n':
+				default:
+					type = LINE_CONTEXT;
+					break;
+				}
+				print_color (out, type, "%s", line);
+			}
```

Comment:
Can be reduced to:
```
switch (line[0]) {
case '+':
	type = LINE_ADDED;
	break;
case '-':
	type = LINE_REMOVED;
	break;
default:
	fwrite (line, (size_t) got, 1, out);
	continue;
}

print_color (out, type, "%s", line);
```

And `enum line_type type;` can be moved to the top of the loop along with `ssize_t got`.